### PR TITLE
BSP-382: Adding structure needed for handling scanned documents

### PR DESCRIPTION
**Pull request by jvictorcf**
### Change description ###
Adding structure needed for handling scanned documents

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bsp/common/model/shared/in/ExceptionRecord.java [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
+import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Getter;
+import lombok.NoArgsConstructor;
 
 import java.util.List;
 
 @Getter
 @Builder
+@AllArgsConstructor
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/bsp/common/model/shared/in/ExceptionRecord.java:13]: whats the reason for need both of these constructors? https://projectlombok.org/features/constructor

File: src/main/java/uk/gov/hmcts/reform/bsp/common/service/transformation/BulkScanFormTransformer.java [hunk 2]
```diff
@@ -29,6 +31,10 @@
         return caseData;
     }
 
+    protected Map<String, Object> transformAdditionalDataFromExceptionRecord(ExceptionRecord exceptionRecord) {
+        return Collections.emptyMap();
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/bsp/common/service/transformation/BulkScanFormTransformer.java:35]: is this unfinished?

File: src/main/java/uk/gov/hmcts/reform/bsp/common/model/shared/in/ExceptionRecord.java (hunk 1 above)
Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/bsp/common/model/shared/in/ExceptionRecord.java:13]: Hi @shaunganley. Jackson needs a no-param constructor and I think we were already using a constructor with all parameters in the tests.

File: src/main/java/uk/gov/hmcts/reform/bsp/common/service/transformation/BulkScanFormTransformer.java (hunk 2 above)
Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/bsp/common/service/transformation/BulkScanFormTransformer.java:35]: No. It's finished.
The default behaviour for this method (when not overridden) is to add no new fields to the caseData (empty map).

**Pull request merged**